addon/components/frost-navigation.js

```diff
@@ -1,48 +1,42 @@
 import Ember from 'ember'
 import layout from '../templates/components/frost-navigation'
 import transitions from 'ember-frost-navigation/transitions/frost-navigation'
-
 const {
   Component,
-  inject,
+  inject: {
+    service
+  },
   computed: {
     alias
   },
-  run
+  run: {
+    scheduleOnce
+  },
+  get,
+  set
 } = Ember
 
 export default Component.extend({
-  frostNavigation: inject.service(),
-  liquidFireTransitions: inject.service(),
+  frostNavigation: service(),
+  liquidFireTransitions: service(),
   classNames: [
-    'frost-navigation',
-    'frost-application-bar'
+    'frost-navigation'
   ],
   layout,
   hook: 'frost-nav',
   categories: alias('frostNavigation.categories'),
+  activeCategory: alias('frostNavigation._activeCategory'),
   init () {
     this._super(...arguments)
 
-    let frostNavigation = this.get('frostNavigation')
-    let liquidFireTransitions = this.get('liquidFireTransitions')
+    let frostNavigation = get(this, 'frostNavigation')
+    let liquidFireTransitions = get(this, 'liquidFireTransitions')
 
-    let ctrl = this.get('targetObject')
+    set(frostNavigation, 'ctrl', get(this, 'targetObject'))
     liquidFireTransitions.map(transitions)
 
-    frostNavigation.set('_controller', ctrl)
-    frostNavigation.addObserver(
-      '_activeCategory',
-      frostNavigation,
-      () => {
-        let active = frostNavigation.get('_activeCategory')
-        if (active) {
-          ctrl.set('activeCategory', active)
-        }
-      }
-    )
     window.addEventListener('popstate', () => {
```

Comment:
Just to clarify - what case is the window popstate listener covering that navCategory dismiss inside the click handler doesn't?